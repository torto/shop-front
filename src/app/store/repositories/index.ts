export * from "./repositories";
export * from "./model";
