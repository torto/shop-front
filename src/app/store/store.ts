import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import repositories from './repositories/repositories'
import repository from './repository/repository'

export const store = configureStore({
  reducer: {
    repositories,
    repository,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
