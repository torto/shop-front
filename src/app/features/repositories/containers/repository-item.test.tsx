import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { Provider } from "react-redux";
import store from "../../../store/repository/repository";
import RepositoryItem from "./repository-item";
import { createStore } from 'redux';

const repositoryMock = {
  repository: {
    data: {
      id: 1,
      name: "name",
      html_url: "http://aa.com",
      description: "description",
      created_at: "2020-10-04T12:04:29.676Z",
      avatar_url: "http://aa.com.image.jps",
      stargazers_count: 1000,
      type: "User",
      login: "torto",
      isStar: false,
    },
  },
};

test("renders component and trigger handleToggleStar", () => {
  const storeMock = createStore(store, { ...repositoryMock } as any);
  storeMock.dispatch = jest.fn();
  
  const {getByRole} = render(
      <Provider store={storeMock}>
      <RepositoryItem id={1} />
    </Provider>
  );
  
  fireEvent.click(getByRole("button"));
  expect(storeMock.dispatch).toHaveBeenCalledTimes(2);
});

test("renders component and trigger handleToggleStar", () => {
    const storeMock = createStore(store, { repository: { data: {} } } as any);
    storeMock.dispatch = jest.fn();
    
  const { asFragment } = render(
    <Provider store={storeMock}>
      <RepositoryItem id={1} />
    </Provider>
  );

  expect(storeMock.dispatch).toHaveBeenCalledTimes(1);
  expect(asFragment()).toMatchSnapshot();

});
