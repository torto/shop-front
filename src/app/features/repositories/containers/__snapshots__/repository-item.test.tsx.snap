// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component and trigger handleToggleStar 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root CardEmpty-card-11 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardHeader-root"
    >
      <div
        class="MuiCardHeader-avatar"
      >
        <span
          class="MuiSkeleton-root MuiSkeleton-circle MuiSkeleton-wave"
          style="width: 40px; height: 40px;"
        />
      </div>
      <div
        class="MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiCardHeader-title MuiTypography-body2 MuiTypography-displayBlock"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
            style="width: 80%; height: 10px; margin-bottom: 6px;"
          />
        </span>
        <span
          class="MuiTypography-root MuiCardHeader-subheader MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
            style="width: 40%; height: 10px;"
          />
        </span>
      </div>
    </div>
    <span
      class="MuiSkeleton-root MuiSkeleton-rect CardEmpty-media-12 MuiSkeleton-wave"
    />
    <div
      class="MuiCardContent-root"
    >
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style="height: 10px; margin-bottom: 6px;"
      />
      <span
        class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style="width: 80%; height: 10px;"
      />
    </div>
  </div>
</DocumentFragment>
`;
